### PR TITLE
Widget: Avoid tracking remove event handlers for classes options

### PR DESCRIPTION
**Pull request by scottgonzalez**
Fixes #15082
Fixes #15095

### CODE REVIEW

**Comment by arschmitz** [comment]
👍

**Comment by scottgonzalez** [comment]
Updated to avoid leaking event listeners.

**Comment by wangchangshan** [comment]
I also find this issue, thanks for your works ~~

**Comment by scottgonzalez** [comment]
We might need to add a `remove` event listener to the element that is having classes added so that we untrack the element when it gets removed. This would catch scenarios like autocomplete menus having items created and destroyed constantly.

**Comment by scottgonzalez** [comment]
Updated to actually work. I confirmed that this fixes all reported issues with the `remove` event and have updated the commit message to reference all tickets.

File: ui/widget.js [hunk 1]
```diff
@@ -493,11 +493,29 @@ $.Widget.prototype = {
 			classes: this.options.classes || {}
 		}, options );
 
+		function bindRemoveEvent() {
+			options.element.each( function( _, element ) {
+				var isTracked = $.map( that.classesElementLookup, function( elements ) {
+					return elements;
+				} )
+					.some( function( elements ) {
```
Comment (arschmitz) [inline ui/widget.js]: This is a single element, so the name should be singular, though element is already taken

Comment (scottgonzalez) [inline ui/widget.js]: This is a collection of elements for a single class.

**Pull request closed without merging**